accumulator/forestdata.go

```diff
@@ -86,7 +118,1054 @@ func (r *ramForestData) close() {
+	if err != nil {
+		return err
+	}
+
+	if m.currentManifestNum > 0 {
+		// Remove old manifest
+		fOldName := fmt.Sprintf("MANIFEST-%06d", m.currentManifestNum)
+		OldFPath := filepath.Join(basePath, fOldName)
+		err = os.Remove(OldFPath)
+		if err != nil {
+			e := fmt.Errorf("ErrOldManifestNotRemoved")
+			return e
+		}
+	}
+
+	return nil
+}
+
+// load loades the manifest from the disk
+func (m *manifest) load(path string) error {
+	curFileName := filepath.Join(path, "CURRENT")
+
+	curFile, err := os.OpenFile(curFileName, os.O_RDONLY, 0666)
+	if err != nil {
+		return err
+	}
+
+	manifestBytes, err := ioutil.ReadAll(curFile)
+	if err != nil {
+		return err
+	}
+
+	maniFName := string(manifestBytes[:])
+
+	maniFilePath := filepath.Join(path, maniFName)
+
+	maniNumString := strings.Replace(
+		maniFName, "MANIFEST-", "", -1)
+
+	// set manifest num
+	m.currentManifestNum, err = strconv.ParseUint(
+		maniNumString, 10, 64)
+	if err != nil {
+		return err
+	}
+
+	maniFile, err := os.Open(maniFilePath)
+	if err != nil {
+		return err
+	}
+
+	// 45 bytes are all that's needed to load except for the locations
+	buf := make([]byte, 45)
+
+	_, err = maniFile.Read(buf)
+	if err != nil {
+		return err
+	}
+
+	// 1. Read forestRows
+	m.forestRows = uint8(buf[0])
+
+	if verbose {
+		fmt.Println("forestRows:", m.forestRows)
+	}
+
+	// 2. Read currentBlockHeight
+	m.currentBlockHeight = int32(binary.LittleEndian.Uint32(buf[1:5]))
+
+	if verbose {
+		fmt.Println("currentBlockHeight:", m.currentBlockHeight)
+	}
+
+	// 3. Read fileNum
+	m.fileNum = binary.LittleEndian.Uint64(buf[5:13])
+	if verbose {
+		fmt.Println("fileNum", m.fileNum)
+	}
+
+	// 4. Read currentBlockHash
+	copy(m.currentBlockHash[:], buf[13:45])
+
+	if verbose {
+		fmt.Println("curBlockH", m.currentBlockHash)
+	}
+
+	var treeBlockRow int
+	// 5. Append locations
+	for {
+		sizeBuf := make([]byte, 4)
+
+		_, err := maniFile.Read(sizeBuf)
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			return err
+		}
+		m.location = append(m.location, []uint64{})
+
+		rowSize := binary.LittleEndian.Uint32(sizeBuf)
+
+		if verbose {
+			fmt.Println("rowsize", rowSize)
+		}
+		rowBytes := make([]byte, rowSize*binary.MaxVarintLen64)
+
+		_, err = maniFile.Read(rowBytes)
+		if err != nil {
+			return err
+		}
+
+		for i := uint32(0); i < rowSize; i++ {
+			start := i * binary.MaxVarintLen64
+			rowToAppend := binary.LittleEndian.Uint64(
+				rowBytes[start : start+binary.MaxVarintLen64])
+			m.location[treeBlockRow] = append(m.location[treeBlockRow], rowToAppend)
+		}
+		treeBlockRow++
+	}
+	if verbose {
+		fmt.Println(m.location)
+	}
+
+	return nil
+}
+
+// treeBlock is a representation of a row 6 utreexo tree.
+// It contains 127 leaves. The leaves can be a UTXO hash or another treeBlock
+type treeBlock struct {
+	// leaves are the utreexo nodes that are either
+	// 1: Hash of a UTXO
+	// 2: Parent hash of two nodes
+	//
+	// Able to hold up to 64 leaves
+	leaves [127]Hash
```

Comment:
have you tried out different block sizes?
might be cool to have a benchmark comparing different sizes.

Reply:
I have not yet. That's also something to try... I should change that 127 to a const variable though